roles/ycsb/tasks/main.yml

```diff
@@ -0,0 +1,208 @@
+---
+- name: Get benchmark state
+  k8s_facts:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: "{{ meta.name }}"
+    namespace: "{{ operator_namespace }}"
+  register: resource_state
+
+- k8s_status:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: "{{ meta.name }}"
+    namespace: "{{ operator_namespace }}"
+    status:
+      state: Building
+      complete: false
+  when: resource_state.resources[0].status.state is not defined
+
+- name: Get benchmark state
+  k8s_facts:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: "{{ meta.name }}"
+    namespace: "{{ operator_namespace }}"
+  register: resource_state
+
+- block:
+
+  - name: setting ycsb_workload_load var
+    set_fact:
+      ycsb_workload_load: "workloada"
+
+  - name: Generate workloads
+    k8s:
+      definition: "{{ lookup('template', 'configmap.yml.j2') | from_yaml }}"
+
+  - name: Load Data Into Database
+    k8s:
+      definition: "{{ lookup('template', 'ycsb_load.yaml') | from_yaml }}"
+
+  - name: Update resource state
+    k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Running Load"
+
+  when: resource_state.resources[0].status.state == "Building" 
+
+- block:
+
+  - name: setting ycsb_workload_load var
+    set_fact:
+      ycsb_workload_load: "workloada"
+  
+  - name: Wait for Load Job to Succeed...
+    k8s_facts:
+      kind: Job
+      api_version: batch/v1
+      name: 'ycsb-data-load-job-{{ ycsb_workload_load }}-{{ trunc_uuid }}'
+      namespace: "{{ operator_namespace }}"
+      label_selectors:
+        - name = 'ycsb-load-{{ trunc_uuid }}'
+    register: ycsb_load_pod
+
+  - name: Update resource state
+    k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Starting First Workload"
+    when: "ycsb_load_pod | json_query('resources[].status.succeeded')"
+ 
+  - name: Update resource state
+    k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Complete"
+    when: workload_args.workloads is undefined
```

Comment:
@jtaleric @aakarshg Not sure if this makes sense or it should just error here. In the case that no workload was given (i.e. only a load) then it just stops saying complete. If preferable I can remove this and it can just error due to the lact of workload list.

Reply:
do you think we can call this state as "Load Complete" ? if its undefined, that'd make super sense.

Reply:
@aakarshg My concern with something other than "Complete" is that it isn't consistent with other workloads when they are "done." Additionally, since it won't be flagged as complete it will continue to fall into the workload loop every time and not just skip over it. Thoughts?